hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/DecayRpcScheduler.java

```diff
@@ -266,6 +268,24 @@ public DecayRpcScheduler(int numLevels, String ns, Configuration conf) {
         DecayRpcSchedulerDetailedMetrics.create(ns);
     decayRpcSchedulerDetailedMetrics.init(numLevels);
 
+    String timeunit = conf.get(CommonConfigurationKeys.RPC_METRICS_TIME_UNIT);
+    TimeUnit tmpTimeUnit;
+    if (StringUtils.isNotEmpty(timeunit)) {
+      try {
+        tmpTimeUnit = TimeUnit.valueOf(timeunit);
+      } catch (IllegalArgumentException e) {
+        LOG.info("Config key {} 's value {} does not correspond to enum values"
+                + " of java.util.concurrent.TimeUnit. Hence default unit"
+                + " {} will be used",
+            CommonConfigurationKeys.RPC_METRICS_TIME_UNIT, timeunit,
+            RpcMetrics.DEFAULT_METRIC_TIME_UNIT);
+        tmpTimeUnit = RpcMetrics.DEFAULT_METRIC_TIME_UNIT;
+      }
+    } else {
+      tmpTimeUnit = RpcMetrics.DEFAULT_METRIC_TIME_UNIT;
+    }
```

Comment:
These lines can be simplified as follows:
```suggestion
    TimeUnit tmpTimeUnit = RpcMetrics.DEFAULT_METRIC_TIME_UNIT;
    if (StringUtils.isNotEmpty(timeunit)) {
      try {
        tmpTimeUnit = TimeUnit.valueOf(timeunit);
      } catch (IllegalArgumentException e) {
        LOG.info("Config key {} 's value {} does not correspond to enum values"
                + " of java.util.concurrent.TimeUnit. Hence default unit"
                + " {} will be used",
            CommonConfigurationKeys.RPC_METRICS_TIME_UNIT, timeunit,
            RpcMetrics.DEFAULT_METRIC_TIME_UNIT);
      }
    }
```

Reply:
Maybe this functionality can be moved to a static method in RpcMetrics, and use them from both DecayRpcScheduler and RpcMetrics.
```
  metricsTimeUnit = RpcMetrics.getMetricsTimeUnitFromConf(conf);
```